engine/flink/avro-util/src/main/scala/pl/touk/nussknacker/engine/avro/schemaregistry/confluent/formatter/ConfluentAvroToJsonFormatterFactory.scala

```diff
@@ -0,0 +1,119 @@
+package pl.touk.nussknacker.engine.avro.schemaregistry.confluent.formatter
+
+import io.circe.generic.semiauto.{deriveDecoder, deriveEncoder}
+import io.circe.{Decoder, Encoder, Json}
+import org.apache.flink.streaming.connectors.kafka.KafkaDeserializationSchema
+import org.apache.kafka.clients.consumer.ConsumerRecord
+import pl.touk.nussknacker.engine.api.CirceUtil
+import pl.touk.nussknacker.engine.api.test.{TestDataSplit, TestParsingUtils}
+import pl.touk.nussknacker.engine.avro.schemaregistry.confluent.ConfluentUtils
+import pl.touk.nussknacker.engine.avro.schemaregistry.confluent.client.ConfluentSchemaRegistryClientFactory
+import pl.touk.nussknacker.engine.kafka.consumerrecord.SerializableConsumerRecord
+import pl.touk.nussknacker.engine.kafka.{KafkaConfig, RecordFormatter, RecordFormatterFactory}
+
+import java.nio.charset.StandardCharsets
+import scala.reflect.ClassTag
+
+class ConfluentAvroToJsonFormatterFactory(schemaRegistryClientFactory: ConfluentSchemaRegistryClientFactory) extends RecordFormatterFactory {
+
+  override def create[K: ClassTag, V: ClassTag](kafkaConfig: KafkaConfig, kafkaSourceDeserializationSchema: KafkaDeserializationSchema[ConsumerRecord[K, V]]): RecordFormatter = {
+
+    val schemaRegistryClient = schemaRegistryClientFactory.create(kafkaConfig)
+    val messageFormatter = new ConfluentAvroMessageFormatter(schemaRegistryClient.client)
+    val createReader = (topic: String) => new ConfluentAvroMessageReader(schemaRegistryClient.client, topic)
+
+    new RecordFormatter {
+
+      override protected def formatRecord(record: ConsumerRecord[Array[Byte], Array[Byte]]): Array[Byte] =
+        doFormatRecord[K, V](kafkaConfig, kafkaSourceDeserializationSchema, messageFormatter)(record)
+
+      override protected def parseRecord(topic: String, bytes: Array[Byte]): ConsumerRecord[Array[Byte], Array[Byte]] =
+        doParseRecord[K, V](kafkaConfig, createReader)(topic, bytes)
+
+      override protected def testDataSplit: TestDataSplit = TestParsingUtils.newLineSplit
+    }
+  }
+
+  /**
+    * Step 1: Deserialize raw kafka event to GenericRecord/SpecificRecord domain.
+    * Step 2: Create Encoders that use ConfluentAvroMessageFormatter to convert avro object to json
+    * Step 3: Encode event's data with schema id's with derived encoder.
+    * @tparam K - key type passed from KafkaAvroSourceFactory, used to determine which datumReaderWriter use (e.g. specific or generic)
+    * @tparam V - value type passed from KafkaAvroSourceFactory, used to determine which datumReaderWriter use (e.g. specific or generic)
+    */
+  protected def doFormatRecord[K: ClassTag, V: ClassTag](kafkaConfig: KafkaConfig,
```

Comment:
Won't be easier to create `ConfluentAvroToJsonFormatter` class instead of passing such many parameters around?

Reply:
Created `ConfluentAvroToJsonFormatter`